apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -493,14 +493,14 @@ export const resourceRouter = router({
       return query.select(["role"]).execute()
     }),
 
-  getAncestryOf: protectedProcedure
+  getAncestryWithSelf: protectedProcedure
```

Comment:
stupid question - but why change this ah? is it to avoid another query so we can generate the full permalink from this procedure alone?

Reply:
Previously, the `resourceStack` was designed to store only ancestor resources, as there was no need to "double click" into folders to view contained resources. We've updated this implementation to also include the current "self" resource within the resourceStack.

Adding "self" to the query enables the "link modal" to accurately display any already linked resources.